### PR TITLE
WIP Idk if any of this sign up page is right

### PR DESCRIPTION
**Pull request by AndrewCNg24**
I made an attempt at this form thing, but is most likely wrong

### CODE REVIEW

File: client/src/component/Sign-Up-Page.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+}
+
+<form>
+    {/* First Name */}
+    <div>
+        <label>First Name</label>
+        <input type="text" name="FirstName" ref="name" placeholder="First Name" />
```
Comment (adriandarian) [inline client/src/component/Sign-Up-Page.jsx]:
```suggestion
        <input type="text" name="FirstName" ref="name" placeholder="First Name" />
```

I believe something is wrong with the `ref`. Like you implemented it incorrectly. Take a look at the [React docs](https://reactjs.org/docs/refs-and-the-dom.html). 

Also you can check that the data is being passed to the super prop using a browser extension called [Reactjs devtools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi).

File: client/src/component/Sign-Up-Page.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+
+class SignUpHacker extends Component {
```
Comment (adriandarian) [inline client/src/component/Sign-Up-Page.jsx]:
```suggestion
class SignUpHacker extends Component {
```

Let's make a centralized `SignUpComponent` that will be the basis for all signups and then we fill more data based on the the previously filled in textfields similar to Typeform, but dynamic across all types of users.

File: client/src/component/Sign-Up-Page.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+
```
Comment (adriandarian) [inline client/src/component/Sign-Up-Page.jsx]: You will need axios to do your post requests to the database through the endpoints we setup.

File: client/src/component/Sign-Up-Page.jsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            gitHub: '',
+            LinkedIn: '',
+            Devpost: '',
+        };
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+}
+
+<form>
```
Comment (adriandarian) [inline client/src/component/Sign-Up-Page.jsx]: Can you attach an image of the form you setup so we can have a visual guide?

File: client/src/component/Sign-Up-Page.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (adriandarian) [reply client/src/component/Sign-Up-Page.jsx]: TBH I do not know exactly how to fix these issues, but this is what I think it means..
image

I think it is asking you to add the `for=name` attribute to the <label> and give the <input> a `id` and `class`

[Resource](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility)

[the remark this answers is not in the document]
Comment (shiv248) [reply client/src/component/Sign-Up-Page.jsx]: @adriandarian I belive that ReactJS throws a compiler error when you leave `<input>` without a closing statement, `</input>` as you showed above. I received the same commit changes from the accesslint bot as @AndrewCNg24 did. I resolved this by changing my `<input/>` statements to `<input></input>`.

**Pull request merged**